src/main/java/io/confluent/idesidecar/restapi/exceptions/ExceptionMappers.java

```diff
@@ -292,7 +292,7 @@ public Response mapApiException(ApiException exception) {
   public Response mapRestClientException(RestClientException exception) {
     var error = io.confluent.idesidecar.restapi.kafkarest.model.Error
         .builder()
-        .errorCode(exception.getStatus())
+        .errorCode(exception.getErrorCode())
```

Comment:
This ensures that the specific error code (like `40401`, `40002` etc) is properly returned in the body for VS Code to differentiate between different types of top-level status codes.